src/effects/nyquist/Nyquist.cpp

```diff
@@ -1064,7 +1064,7 @@ bool NyquistEffect::Process(EffectInstance &, EffectSettings &settings)
 
    // Has rug been pulled from under us by some effect done within Nyquist??
    if (!bOnePassTool && (nEffectsSoFar == EffectOutputTracks::nEffectsDone)) {
-      if (success)
+      if (success && mType != EffectTypeAnalyze && mType != EffectTypeNone)
```

Comment:
Not sure if this is exhaustive. This means that `EffectTypeTool` or `EffectTypeHidden` have their outputs commit, but especially in the latter case, I'm not sure that this is right.
Maybe this PR is worth more effort, in providing a virtual `bool Nyquist::ModifiesAudio()`  method, at least to improve clarify and maybe to realize that there are effects which aren't of the type analyze or none that don't want their output to be processed. What do you think @Paul-Licameli ?

Reply:
No, I think this change is good enough.

`EffectTypeHidden` is just StereoToMono which shares code with effects but is specially put in a different menu (Tracks).

`EffectTypeTool` includes the Nyquist prompt, in which you can type Lisp interactively and expect it to operate just as if a Nyquist effect.  It also includes Nyquist scripts that implement macros, but I'm not concerned about that case.

Reply:
Ok, so the Nyquist prompt will still render stretching, which is fine. I couldn't reach this place splitting stereo-to-mono the usual way, and haven't found a Nyquist way of doing that.